addons/mrp/models/mrp_production.py

```diff
@@ -237,6 +235,26 @@ def _set_date_planned_start(self):
     def _set_date_planned_finished(self):
         self.move_finished_ids.write({'date_expected': self.date_planned_finished})
 
+    @api.depends('move_raw_ids.delay_alert_date')
+    def _compute_delay_alert_date(self):
+        delay_alert_date_data = self.env['stock.move'].read_group([('id', 'in', self.move_raw_ids.ids), ('delay_alert_date', '!=', False)], ['delay_alert_date:max'], 'raw_material_production_id')
+        delay_alert_date_data = {data['raw_material_production_id'][0]: data['delay_alert_date'] for data in delay_alert_date_data}
+        for production in self:
+            production.delay_alert_date = delay_alert_date_data.get(production.id, False)
+
+    def _compute_json_popover(self):
+        for production in self:
+            production.json_popover = json.dumps({
+                'popoverTemplate': 'mrp.ProductionPopover',
+                'model': 'mrp.production',
+                'delay_alert_date': format_datetime(self.env, production.delay_alert_date, dt_format=False) if production.delay_alert_date else False,
+                'late_elements': [{
+                        'id': late_move.id,
+                        'name': late_move.display_name,
+                    } for late_move in production.move_raw_ids.filtered(lambda m: m.delay_alert_date).move_orig_ids.production_id
```

Comment:
this only handles child mo late for the parent mo
you also have
- po late for the components of an mo
- pbm late for its mo